webknossos/webknossos/annotation/annotation.py

```diff
@@ -434,6 +436,52 @@ def _write_to_zip(self, zipfile: ZipFile) -> None:
                 layer_content,
             )
 
+    def get_annotation_dataset(self) -> Dataset:
+        """Returns a streamed dataset, incorporating fallback layers and mappings."""
+        from webknossos.client.context import _get_context
+
+        if self.annotation_id is None:
+            raise ValueError(
+                "The annotation_id is not set, cannot get the corresponding dataset."
+            )
+
+        context = _get_context()
+        token: Optional[str]
+        if self.organization_id is None:
+            token = context.required_token
+            organization_id = context.organization_id
+        else:
+            token = context.token
+            organization_id = self.organization_id
+            # avoid requiring authentication
+            if token is not None:
+                if organization_id != context.organization_id:
+                    warnings.warn(
+                        "The annotation used in with get_annotation_dataset "
+                        + "specifies a different organization id than the current context. "
+                        + f"The annotation uses {organization_id}, the context {context.organization_id}.",
+                        RuntimeWarning,
+                    )
+
+        dataset_info_response = dataset_info.sync_detailed(
+            organization_name=organization_id,
+            data_set_name=self.dataset_name,
+            client=context.generated_client,
+        )
+        assert dataset_info_response.status_code == 200, dataset_info_response
+        parsed = dataset_info_response.parsed
+        assert parsed is not None
+
+        # token = sharing_token or wk_context.datastore_token
+
+        datastore_url = parsed.data_store.url
+
+        zarr_path = UPath(
+            f"{datastore_url}/data/annotations/zarr/{self.annotation_id}/",
```

Comment:
the entry point is always at the datastore and never at tracing store, right? @fm3 can you confirm this?